src/bindings/g_object/object.cr

```diff
@@ -33,6 +39,290 @@ module GObject
+          {% verbatim do %}
+            {% begin %}
+              {% instance_vars = @type.instance_vars.select(&.annotation(GObject::Property)) %}
+
+              @@_g_param_specs = Pointer(LibGObject::ParamSpec*).malloc({{ instance_vars.size }})
+              {% for var, i in instance_vars %}
+                {% property = var.annotation(GObject::Property) %}
+                name = {{ var.name.gsub(/\_/, "-").stringify }}.to_unsafe
+                nick = {{ property["nick"] }}.try(&.to_unsafe) || Pointer(LibC::Char).null
+                blurb = {{ property["blurb"] }}.try(&.to_unsafe) || Pointer(LibC::Char).null
+                {% other_args = property.named_args.to_a.reject { |arg| ["nick", "blurb"].includes?(arg[0].stringify) } %}
+
+                flags = GObject::ParamFlags::StaticName | GObject::ParamFlags::StaticBlurb | GObject::ParamFlags::ExplicitNotify
+                flags |= GObject::ParamFlags::Deprecated unless {{ !!var.annotation(Deprecated) }}
+                flags |= GObject::ParamFlags::Readable if {{ @type.has_method?(var.name.stringify) }}
+                flags |= GObject::ParamFlags::Writable if {{ @type.has_method?("#{var.name}=") }}
+
+                # Finally register the type to GLib.
+                # The given varible name has its underscores converted to dashes.
+                pspec = GObject.create_param_spec({{ var.type }}, name, nick, blurb, flags, {{ other_args.map { |tuple| "#{tuple[0]}: #{tuple[1]}".id }.splat }})
+                @@_g_param_specs[{{ i }}] = pspec.as(LibGObject::ParamSpec*)
+                LibGObject.g_object_class_install_property(klass, {{ i + 1 }}, pspec)
+              {% end %}
+            {% end %}
+          {% end %}
+        end
+
+        # :nodoc:
+        def unsafe_do_get_property(property_id : UInt32, gvalue : Void*, param_spec : Void*) : Nil
+          {% verbatim do %}
+            {% begin %}
+              {% instance_vars = @type.instance_vars.select(&.annotation(GObject::Property)) %}
+
+              case property_id
+              {% for var, i in instance_vars %}
+                {% if @type.has_method?(var.name.stringify) %}
+                  when {{ i + 1 }}
+                    GObject::Value.set_g_value(gvalue.as(LibGObject::Value*), self.{{ var }})
+                {% end %}
+              {% end %}
+              end
+            {% end %}
+          {% end %}
+        end
+
+        # :nodoc:
+        def unsafe_do_set_property(property_id : UInt32, gvalue : Void*, param_spec : Void*) : Nil
+          {% verbatim do %}
+            {% begin %}
+              {% instance_vars = @type.instance_vars.select(&.annotation(GObject::Property)) %}
+
+              case property_id
+              {% for var, i in instance_vars %}
+                {% if @type.has_method?("#{var.name}=") %}
+                  when {{ i + 1 }}
+                    raw = GObject::Value.raw(GObject.fundamental_g_type({{ var.type }}), gvalue)
```

Comment:
GValue doesn't need to be used here neither, so one less heap allocation :tada:

Reply:
I think there is no heap allocation here, I added a class method.